terraform/modules/networking/route_table.tf

```diff
@@ -53,11 +53,11 @@ resource "aws_route" "nat_gateway_to_internet" {
 # Coming back the Internet Gateway will route traffic to the NAT Gateway - that's the default, all good
 # Then the NAT Gateway should route traffic destined for firewalled subnets back through the firewall first
 resource "aws_route" "nat_gateway_back_to_firewall" {
-  for_each = { for subnet in local.firewalled_subnets : subnet.tags.Name => subnet }
+  count = length(local.firewall_config)
```

Comment:
Most of the configs apply to more than one subnet, so I don't think this will work.

We might just have to take the "subnets" property out of firewall_config, and have it be a separate list.